swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt

```diff
@@ -128,23 +91,31 @@ abstract class ScheduledAgent(
     runningTask?.cancel(true)
   }
 
-  override fun process(workConfiguration: WorkConfiguration, onCompleteCallback: () -> Unit) {
-    val action = getAction()
-    val handlerAction: (handler: ResourceTypeHandler<*>) -> Unit = {
-      when (action) {
-        Action.MARK -> it.mark(workConfiguration, onCompleteCallback)
-        Action.NOTIFY -> it.notify(workConfiguration, onCompleteCallback)
-        Action.DELETE -> it.delete(workConfiguration, onCompleteCallback)
-        else -> log.warn("Unknown action {}", action.name)
+  /**
+   * Takes work [WorkConfiguration] off the [WorkQueue], acquires a lock and
+   * dispatches the work to a suitable handler for an [Action] for processing
+   */
+  override fun process() {
```

Comment:
The work queue is loaded with the work configuration, which is `provider:account:region:resourceType`. This means that if snapshots and images are both enabled for `aws:test:us-east-1` then we have two work configuration entries:
[`aws:test:us-east-1:image`,
 `aws:test:us-east-1:snapshot`]

This function is scheduled for every agent (mark/notify/delete) every `intervalSeconds`. But, since the work queue is only size 2, and gets refilled only on a fixed delay (`swabbie.queue.monitorIntervalMs`), we do a lot less work than I'd expect. (2 operations instead of all 6 operations - mark notify delete for each work configuration - in the interval)

I would like for this work queue to have each type of work in it (mark/notify/delete for each work configuration) and then each instance picks this up one at a time, and the queue gets refilled after `intervalSeconds` only if the queue is empty.

I'd expect swabbie to work like this:
- work queue is populated with each action for each work configuration, which would look like
[`MARK` `aws:test:us-east-1:image`,
 `MARK` `aws:test:us-east-1:snapshot`,
 `NOTIFY` `aws:test:us-east-1:image`,
 `NOTIFY` `aws:test:us-east-1:snapshot`
 `DELETE` `aws:test:us-east-1:image`,
 `DELETE` `aws:test:us-east-1:snapshot` ]
- each instance picks one item of work off the queue, and does it.
- when that instance is done with that item, it picks another item off the queue
- when the queue is empty, the instance does nothing
- every `intervalSeconds` (for each agent type?) the queue is refilled with those work items

This way, all the work is done each interval as we would expect, but each instance does only one thing at a time.

Reply:
Missed that, agreed should be 6 work items. Will update that part.

So is your desire to see an instance to continuously do work as long as the queue has work?
I think that's reasonable. @emjburns

Reply:
Ya, I think that makes sense behavior-wise, so all the work gets done. Then we don't have to deal with scheduling things!